includes/Core/Authentication/Token.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Authentication\Token
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Authentication;
+
+use Google\Site_Kit\Core\Storage\User_Options_Interface;
+use Google\Site_Kit\Core\Storage\Encrypted_User_Options;
+use Google\Site_Kit\Core\Authentication\Clients\OAuth_Client;
+
+/**
+ * Class representing the OAuth token for a user.
+ *
+ * This includes the access token, its creation and expiration data, and the refresh token.
+ * This class is compatible with `Google\Site_Kit\Core\Storage\User_Setting`, as it should in the future be adjusted
+ * so that the four pieces of data become a single user setting.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+final class Token {
+
+	/**
+	 * User_Options_Interface implementation.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Options_Interface
+	 */
+	protected $user_options;
+
+	/**
+	 * Encrypted_User_Options instance.
+	 *
+	 * @since 1.0.0
+	 * @var Encrypted_User_Options
+	 */
+	private $encrypted_user_options;
+
+	/**
+	 * User_Setting constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param User_Options_Interface $user_options User_Options_Interface instance.
+	 */
+	public function __construct( User_Options_Interface $user_options ) {
```

Comment:
Good point - however, why is that though? Is there a good reason or was that just an oversight? Looking at it now, I think it might be most appropriate to change `Encrypted_User_Options` to expect `User_Options_Interface` instead?

Reply:
`User_Options` and `Encrypted_User_Options` both implement the `User_Options_Interface`. So if we changed `Encrypted_User_Options` to accept `User_Options_Interface` instead, then it could technically be instantiated with an instance of `Encrypted_User_Options` which of course wouldn't work, hence breaking the contract.

So in this case, the `User_Options_Interface` abstracts the storage implementation which includes (un)encrypted data, but `Encrypted_User_Options` needs to read and write "raw" values. As lower-level classes, I think this is okay given our current architecture.

To your point, I think we would need new interfaces for plaintext and encrypted storage in order to allow these dependencies to be passed fulfilling an interface rather than a concrete implementation.

Reply:
I think this is the only thing left to address.